### PR TITLE
 sim/rptun: support multi(>2) core interactive by share memory

### PR DESCRIPTION
**Pull request by Donny9**
## Summary
Support multi >2 core interactive by share memory in simulation.
Example of relationships between the three processes:
AP->BP
AP->CP
BP->CP
## Impact
Support multi-core heterogeneous simulation.
## Testing
daily test.

### CODE REVIEW

**Comment by Donny9** [comment]
This patch depends on #3726 3726 and #3720, please review and merge them firsty .Thank you!

**Comment by acassis** [comment]
> This patch depends on #3726 3726 and #3720, please review and merge them firsty .Thank you!

Hi @Donny9 I think you repeated #3726, it should be other PR.

**Comment by Donny9** [comment]
> 
> 
> > This patch depends on #3726 3726 and #3720, please review and merge them firsty .Thank you!
> 
> Hi @Donny9 I think you repeated #3726, it should be other PR.

I modify this patch base #3726,  #3726 is a modification of the original code.

File: boards/sim/sim/sim/Kconfig [hunk 1]
```diff
@@ -10,6 +10,10 @@ config EXAMPLES_TOUCHSCREEN_BGCOLOR
 	default 0x007b68ee
 	depends on EXAMPLES_TOUCHSCREEN
 
+config SIM_RPTUN_MASTER
+	bool "Remote Processor Tunneling Role"
+	depends on RPTUN
```
Comment (gustavonihei) [inline boards/sim/sim/sim/Kconfig:15]: Add the default value for this boolean option.

Comment (Donny9) [reply boards/sim/sim/sim/Kconfig:15]: > Add the default value for this boolean config.
Done. thank you.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**